### PR TITLE
Enhanced features

### PR DESCRIPTION
**Pull request by sainho93**
1. For issue #195: highlighting for whole room or zone
2. For issue #197: added tab for saving jpscore path
2. For issue #198: fixed exiting_crossing into 8 for floorfied

### CODE REVIEW

File: src/widgets/roomlistwidget.cpp [hunk 1]
```diff
@@ -440,4 +445,89 @@ void RoomListWidget::deleteZoneButtonClicked()
...
+    if(item == nullptr)
+        return;
 
+    view->unmark_all_lines();
+
+    auto *zone = getCurrentZone(item);
+
+    if(zone->getType() == Platform)
+    {
+        for(JPSTrack *track : zone->getTrackList())
```
Comment (zeroset) [inline src/widgets/roomlistwidget.cpp]: const ?

File: src/widgets/roomlistwidget.cpp [hunk 2]
```diff
@@ -440,4 +445,89 @@ void RoomListWidget::deleteZoneButtonClicked()
...
+        {
+            view->markLine(track->getLine());
+        }
+    } else
+    {
+        for(jpsLineItem *wall : zone->get_listWalls())
```
Comment (zeroset) [inline src/widgets/roomlistwidget.cpp]: const?

File: src/widgets/roomlistwidget.cpp [hunk 3]
```diff
@@ -440,4 +445,89 @@ void RoomListWidget::deleteZoneButtonClicked()
...
+        {
+            view->markLine(wall);
+        }
+    }
+
+    // Mark crossing
+    for(jpsCrossing *crossing : zone->getEnterAndExitList())
```
Comment (zeroset) [inline src/widgets/roomlistwidget.cpp]: const

File: src/widgets/roomlistwidget.cpp [hunk 4]
```diff
@@ -440,4 +445,89 @@ void RoomListWidget::deleteZoneButtonClicked()
...
+    {
+        view->markLine(crossing->get_cLine());
+    }
+
+    // Mark transitions in the subroom
+    for(jpsTransition *transition : data->getTransitionInSubroom(zone))
```
Comment (zeroset) [inline src/widgets/roomlistwidget.cpp]: const

**Comment by sainho93** [comment]
@zeroset I'm not sure here if it should use "const" or not. If using const, so it's a pointer to const, it means pointer track shouldn't be changed, but in fact it should change its QPen by markLine() function.

**Comment by zeroset** [comment]
Sorry my fault, thought MarkLine doesn't edit.

**Pull request merged**